### ISSUE TITLE
sphinx-build hard-codes the python path into the exe

### ISSUE THREAD

**Issue by dylanh724** [sphinx-doc/sphinx#12342]
### Describe the bug

If you actually open `sphinx-build.exe`, you can see the shebang:

![image](https://github.com/sphinx-doc/sphinx/assets/8840024/ffdb8356-86e7-4f9a-8370-42459b182929)

Shouldn't it just use whatever python exe is in PATH? Why is this hard-coded? This is... strange. This is even stranger that it's own install path isn't here. I don't have a Python install at this path, nor do I have an env var that shows this path.

### How to Reproduce

1. Install Python 3.10 via Microsoft Store
2. `pip install sphinx-build`
3. `sphinx-build`

### Environment Information

```text
Win11:

> sphinx-build --bug-report
Fatal error in launcher: Unable to create process using '"C:\Program Files\Python310\python.exe"  "C:\Users\dylan\AppData\Roaming\Python\Python310\Scripts\sphinx-build.exe" --bug-report': The system cannot find the file specified.


You can see that it's being called from the correct python path (appdata), but the exe hard-codes a bad path for some reason. I ensured env vars are clean.
```

### Additional context

I'm a new user to sphinx. I had instructions to do the following in the colleagues readme:

> Using [Chocolatey](https://chocolatey.org/): `choco install make --yes`
> Using [pip](https://www.dataquest.io/blog/install-pip-windows/): `pip install sphinx sphinx-tabs recommonmark sphinx_rtd_theme`

**Comment by chrisjsewell** [comment]
Hey @dylanh724, sphinx uses the standard Python declaration for scripts: https://github.com/sphinx-doc/sphinx/blob/c41aab829d667a6fb5c9e6e8daf2d4b47b7ddebe/pyproject.toml#L106-L107

So if there is a problem it would likely lie with [flit](https://github.com/pypa/flit) or [pip](https://github.com/pypa/pip)
(I've never encountered this problem, but then I don't use windows)

Alternatively, you can more specifically target the correct Python executable using: `python -m sphinx.cmd.build `

**Comment by dylanh724** [comment]
> Hey @dylanh724, sphinx uses the standard Python declaration for scripts:
> 
> https://github.com/sphinx-doc/sphinx/blob/c41aab829d667a6fb5c9e6e8daf2d4b47b7ddebe/pyproject.toml#L106-L107
> 
> So if there is a problem it would likely lie with [flit](https://github.com/pypa/flit) or [pip](https://github.com/pypa/pip) (I've never encountered this problem, but then I don't use windows and also no idea what chocolatey is)
> 
> Alternatively, you can more specifically target the correct Python executable using: `python -m sphinx.cmd.build`, let me know if that works

Got it -- 
```bash
 WARNING: The scripts sphinx-apidoc.exe, sphinx-autogen.exe, sphinx-build.exe and sphinx-quickstart.exe are installed in 'C:\Users\dylan\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts' which is not on PATH.
  Consider adding this directory to PATH
```

It turns out that the Microsoft Store variant of Python does *not* include /Scripts to path (strange since they add python/pip to path).

Since they have a really wild path/to (you can't even view the dir in Windows - it's encrypted *from yourself* with access denied if you attempt to browse), this warning is the *only* chance to find where the correct path/to/Scripts/ actually is.

Windows...

This is the *most official* way to get Python these days, so you'll probably start seeing more tickets like this @chrisjsewell - thanks for the speedy response.

**Comment by dylanh724** [comment]
## TL;DR

Windows users that installed from Microsoft Store should add the following path to their **user** env var:

`C:\Users\{user}\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts`

⚠️Replacing `{user}`

**Comment by dylanh724** [comment]
^ @chrisjsewell Hmm... you know, it may be worth including this in the Windows install section if they get a command not found err. If this isn't a common issue yet, it will be.

**Comment by chrisjsewell** [comment]
@zooba that's exactly what I already said 😅 https://github.com/sphinx-doc/sphinx/issues/12342#issuecomment-2089402544

**Comment by dylanh724** [comment]
Perhaps this should be in the install docs? 

> pip install -U sphinx 

> After installation, type sphinx-build --version on the command prompt. 

Eg, add the prefix here.
